src/OrchardCore/OrchardCore.Environment.Shell.Abstractions/Builders/ShellContext.cs

```diff
@@ -45,7 +49,10 @@ public IServiceScope EnterServiceScope()
                 throw new InvalidOperationException("Can't use EnterServiceScope on a released context");
             }
 
-            return new ServiceScopeWrapper(ServiceProvider.CreateScope());
+            // Prevent the context from being released until the end of the scope
+            Interlocked.Increment(ref _refCount);
```

Comment:
Hmm, maybe we could increment it in the wrapper where it is also decremented, or having 2 shellContext methods for scope starting / ending that the wrapper would call.